src/core/map.js

```diff
@@ -1194,11 +1226,14 @@ geo.map = function (arg) {
    * Cancel any existing transition.  The transition will send a cancel event
    * at the next animation frame, but no further activity occurs.
    *
+   * @param {string} source option cause of the cancel.
```

Comment:
Yes, `source` is optional  The description should have read 'optional' (not option).  It falls back to the a previous value (if it was set), and that falls back to an empty string.

I've used it as `<method>.<action>`, but that isn't strictly required.  I'll add to the comment that it is preferred.